### ISSUE TITLE
route_path helper not available to overrides

### ISSUE THREAD

**Issue by techgique** [CDRH/orchid#290]
Will and Erin have both recently worked on overrides for search interface partials and run into a problem with the code copied from Orchid causing `undefined local variable or method route_path` errors. An example of this is trying to override the [search box partial](https://github.com/CDRH/orchid/blob/v3.1.1/app/views/items/_search_box.html.erb).

In this template, `prefix_path`'s argument is `route_path`. Within the Orchid Rails engine context, this is calling a helper method: https://github.com/CDRH/orchid/blob/v3.1.1/app/helpers/orchid/application_helper.rb#L308 But this helper method isn't available when overridden from the main Rails app.

We've gotten around this by replacing `route_path` with just passing the default string value that the method returns `"search_path"` but this is confusing when creating overrides and would break use of the `@route_path` value it can return as well. Want to figure out how to make Orchid's helpers like this available to the main app as well.

**Comment by wkdewey** [comment]
I tried to replicate this in my Tibetan Monasteries app and I could access route_path from that partial, it was correctly defined as "search_path".

**Comment by wkdewey** [comment]
I don't think this has come up recently, especially not with new Orchid changes. Doesn't seem high priority, but also isn't "fixed" so we could leave it open.
